### PR TITLE
Solving most of the memory leak, simplifying and some other small bugs

### PR DESCRIPTION
**Pull request by GuillermoCasas**
The memory seems to leak only when running the case twice; after the second time the memory remains stable. More work is in order but the main issue seems to be solved thanks to the addition of more weakrefs to avoid leaking due to python circular references

### CODE REVIEW

File: applications/swimming_DEM_application/python_scripts/swimming_DEM_procedures.py [hunk 1]
```diff
@@ -456,7 +457,7 @@ def __init__(self,
                  rigid_faces_model_part,
                  mixed_model_part):
 
-        self.gid_io                 = gid_io
+        self.gid_io                 = weakref.proxy(gid_io)
```
Comment (roigcarlo) [inline applications/swimming_DEM_application/python_scripts/swimming_DEM_procedures.py:460]: That leak was inside gid_io class (c++ code), it was already fixed in #1074

Comment (GuillermoCasas) [reply applications/swimming_DEM_application/python_scripts/swimming_DEM_procedures.py:460]: Hello, I know about it. I ran the DEM case I had and it worked without leaking, however, the leak remained in my swimming_DEM_app cases (I double checked this to be the case when running the case with the master version). After adding a few weakrefs (which I think make sense anyways) the leak has vanished.

Comment (roigcarlo) [reply applications/swimming_DEM_application/python_scripts/swimming_DEM_procedures.py:460]: Oh, ok, I thought that would have solve all problems with the gid_io, seems i was wrong sorry

**Comment by maceligueta** [comment]
I pushed some fixes.

**Comment by GuillermoCasas** [comment]
@maceligueta  That's fine: it still works. Can you approve?

**Review by maceligueta — APPROVED** [review]

**Pull request merged**